tests/back/api/test_global_explanation_api.py

```diff
@@ -0,0 +1,204 @@
+
+
+@pytest.fixture(scope="module", name="run_id_1")
+def fixture_run_id_1(session: sessionmaker, experiment_id: int):
+    db = session()
+    run = Run(
+        experiment_id=experiment_id,
+        model_name="RandomForestClassifier",
+        parameters={},
+        name="Run",
+    )
+    db.add(run)
+    db.commit()
+    db.refresh(run)
+
+    yield run.id
+
+    db.delete(run)
+    db.commit()
+    db.close()
+
+
+@pytest.fixture(scope="module", name="run_id_2")
+def fixture_run_id_2(session: sessionmaker, experiment_id: int):
+    db = session()
+    run = Run(
+        experiment_id=experiment_id,
+        model_name="SVC",
+        parameters={},
+        name="Run",
+    )
+    db.add(run)
+    db.commit()
+    db.refresh(run)
+
+    yield run.id
+
+    db.delete(run)
+    db.commit()
+    db.close()
+
+
+def test_create_global_explanation(client: TestClient, run_id_1: int, run_id_2):
+    response = client.post(
+        "/api/v1/global-explanation/",
+        json={
+            "name": "test_1",
+            "run_id": run_id_1,
+            "explainer_name": "PartialDependence",
+            "parameters": {"grid_resolution": 100, "percentiles": [0.1, 0.6]},
+        },
+    )
+    assert response.status_code == 201, response.text
+
+    response = client.post(
+        "/api/v1/global-explanation/",
+        json={
+            "name": "test_2",
+            "run_id": run_id_1,
+            "explainer_name": "PartialDependence",
+            "parameters": {"grid_resolution": 50, "percentiles": [0.1, 0.2]},
+        },
+    )
+    assert response.status_code == 201, response.text
+
+    response = client.post(
+        "/api/v1/global-explanation/",
+        json={
+            "name": "test_3",
+            "run_id": run_id_2,
+            "explainer_name": "PartialDependence",
+            "parameters": {"grid_resolution": 50, "percentiles": [0.1, 0.55]},
+        },
+    )
+    assert response.status_code == 201, response.text
+
+
+def test_get_global_explanation_by_id(
+    client: TestClient, run_id_1: int, run_id_2: int, dataset_id: int
+):
+    response = client.get("/api/v1/global-explanation/1")
+    assert response.status_code == 200, response.text
+    data = response.json()
+    assert data["name"] == "test_1"
+    assert data["run_id"] == run_id_1
+    assert data["explainer_name"] == "PartialDependence"
+    assert data["parameters"] == {"grid_resolution": 100, "percentiles": [0.1, 0.6]}
+
+    response = client.get("/api/v1/global-explanation/3")
+    assert response.status_code == 200, response.text
+    data = response.json()
+    assert data["name"] == "test_3"
+    assert data["run_id"] == run_id_2
+    assert data["explainer_name"] == "PartialDependence"
+    assert data["parameters"] == {"grid_resolution": 50, "percentiles": [0.1, 0.55]}
+
+
+def test_get_explainers(
+    client: TestClient, run_id_1: int, run_id_2: int, dataset_id: int
+):
+    response = client.get("/api/v1/global-explanation/")
```

Comment:
Same here about the independence of the tests: you should create them here.

Reply:
Also, if you make the change that /global-explainers brings you only the explanations associated with a run, then this version should not work.